Adopt smart pointers in FrameLoader

cdumez opened the pull request:
#### af1e21c6dbceb40a1b3c4b55578e614bbf28ed22
<pre>
Adopt smart pointers in FrameLoader
<a href="https://bugs.webkit.org/show_bug.cgi?id=263796">https://bugs.webkit.org/show_bug.cgi?id=263796</a>

Reviewed by Ryosuke Niwa.

* Source/WebCore/html/HTMLFormElement.cpp:
(WebCore::HTMLFormElement::submitIfPossible):
* Source/WebCore/html/HTMLFrameOwnerElement.h:
(WebCore::Frame::protectedOwnerElement const):
* Source/WebCore/inspector/InspectorFrontendClientLocal.cpp:
(WebCore::InspectorFrontendClientLocal::openURLExternally):
* Source/WebCore/loader/FormState.cpp:
(WebCore::FormState::protectedSourceDocument const):
* Source/WebCore/loader/FormState.h:
* Source/WebCore/loader/FrameLoadRequest.cpp:
(WebCore::FrameLoadRequest::FrameLoadRequest):
* Source/WebCore/loader/FrameLoadRequest.h:
(WebCore::FrameLoadRequest::FrameLoadRequest):
* Source/WebCore/loader/FrameLoader.cpp:
(WebCore::ForbidPromptsScope::ForbidPromptsScope):
(WebCore::ForbidPromptsScope::~ForbidPromptsScope):
(WebCore::ForbidSynchronousLoadsScope::ForbidSynchronousLoadsScope):
(WebCore::ForbidSynchronousLoadsScope::~ForbidSynchronousLoadsScope):
(WebCore::FrameLoader::FrameProgressTracker::FrameProgressTracker):
(WebCore::FrameLoader::FrameProgressTracker::~FrameProgressTracker):
(WebCore::FrameLoader::FrameProgressTracker::progressStarted):
(WebCore::FrameLoader::FrameProgressTracker::progressCompleted):
(WebCore::FrameLoader::frame const):
(WebCore::FrameLoader::protectedFrame const):
(WebCore::FrameLoader::detachFromAllOpenedFrames):
(WebCore::FrameLoader::init):
(WebCore::FrameLoader::initForSynthesizedDocument):
(WebCore::FrameLoader::pageID const):
(WebCore::FrameLoader::frameID const):
(WebCore::FrameLoader::setDefersLoading):
(WebCore::FrameLoader::shouldUpgradeRequestforHTTPSOnly const):
(WebCore::FrameLoader::changeLocation):
(WebCore::FrameLoader::submitForm):
(WebCore::FrameLoader::stopLoading):
(WebCore::FrameLoader::stop):
(WebCore::FrameLoader::closeURL):
(WebCore::FrameLoader::didOpenURL):
(WebCore::FrameLoader::didExplicitOpen):
(WebCore::FrameLoader::clear):
(WebCore::FrameLoader::receivedFirstData):
(WebCore::FrameLoader::didBeginDocument):
(WebCore::FrameLoader::finishedParsing):
(WebCore::FrameLoader::allChildrenAreComplete const):
(WebCore::FrameLoader::allAncestorsAreComplete const):
(WebCore::FrameLoader::checkCompleted):
(WebCore::FrameLoader::checkCompletenessNow):
(WebCore::FrameLoader::checkCallImplicitClose):
(WebCore::FrameLoader::loadURLIntoChildFrame):
(WebCore::FrameLoader::loadArchive):
(WebCore::FrameLoader::outgoingReferrer const):
(WebCore::FrameLoader::outgoingOrigin const):
(WebCore::FrameLoader::checkIfFormActionAllowedByCSP const):
(WebCore::FrameLoader::setOpener):
(WebCore::FrameLoader::provisionalLoadStarted):
(WebCore::FrameLoader::updateFirstPartyForCookies):
(WebCore::FrameLoader::setFirstPartyForCookies):
(WebCore::FrameLoader::loadInSameDocument):
(WebCore::FrameLoader::completed):
(WebCore::FrameLoader::started):
(WebCore::FrameLoader::prepareForLoadStart):
(WebCore::FrameLoader::loadFrameRequest):
(WebCore::FrameLoader::isNavigationAllowed const):
(WebCore::FrameLoader::loadURL):
(WebCore::FrameLoader::defaultSubstituteDataForURL):
(WebCore::FrameLoader::load):
(WebCore::FrameLoader::loadWithDocumentLoader):
(WebCore::FrameLoader::reloadWithOverrideEncoding):
(WebCore::FrameLoader::reload):
(WebCore::FrameLoader::stopAllLoaders):
(WebCore::FrameLoader::stopForBackForwardCache):
(WebCore::FrameLoader::stopForUserCancel):
(WebCore::FrameLoader::setDocumentLoader):
(WebCore::FrameLoader::setPolicyDocumentLoader):
(WebCore::FrameLoader::setProvisionalDocumentLoader):
(WebCore::FrameLoader::setState):
(WebCore::FrameLoader::provisionalLoadFailedInAnotherProcess):
(WebCore::FrameLoader::commitProvisionalLoad):
(WebCore::FrameLoader::transitionToCommitted):
(WebCore::FrameLoader::closeOldDataSources):
(WebCore::FrameLoader::willRestoreFromCachedPage):
(WebCore::FrameLoader::open):
(WebCore::FrameLoader::subframeIsLoading const):
(WebCore::FrameLoader::subresourceCachePolicy const):
(WebCore::FrameLoader::handleLoadFailureRecovery):
(WebCore::FrameLoader::checkLoadCompleteForThisFrame):
(WebCore::FrameLoader::setOriginalURLForDownloadRequest):
(WebCore::FrameLoader::didReachLayoutMilestone):
(WebCore::FrameLoader::didFirstLayout):
(WebCore::FrameLoader::restoreScrollPositionAndViewStateSoon):
(WebCore::FrameLoader::didReachVisuallyNonEmptyState):
(WebCore::FrameLoader::detachChildren):
(WebCore::FrameLoader::closeAndRemoveChild):
(WebCore::FrameLoader::checkLoadComplete):
(WebCore::FrameLoader::numPendingOrLoadingRequests const):
(WebCore::FrameLoader::userAgent const):
(WebCore::FrameLoader::navigatorPlatform const):
(WebCore::FrameLoader::frameDetached):
(WebCore::FrameLoader::detachFromParent):
(WebCore::FrameLoader::updateRequestAndAddExtraFields):
(WebCore::FrameLoader::scheduleRefreshIfNeeded):
(WebCore::FrameLoader::loadPostRequest):
(WebCore::FrameLoader::loadResourceSynchronously):
(WebCore::FrameLoader::receivedMainResourceError):
(WebCore::FrameLoader::continueFragmentScrollAfterNavigationPolicy):
(WebCore::FrameLoader::shouldPerformFragmentNavigation):
(WebCore::FrameLoader::scrollToFragmentWithParentBoundary):
(WebCore::FrameLoader::shouldClose):
(WebCore::FrameLoader::dispatchUnloadEvents):
(WebCore::FrameLoader::dispatchBeforeUnloadEvent):
(WebCore::FrameLoader::executeJavaScriptURL):
(WebCore::FrameLoader::continueLoadAfterNavigationPolicy):
(WebCore::FrameLoader::continueLoadAfterNewWindowPolicy):
(WebCore::FrameLoader::loadedResourceFromMemoryCache):
(WebCore::FrameLoader::shouldInterruptLoadForXFrameOptions):
(WebCore::FrameLoader::shouldTreatURLAsSameAsCurrent const):
(WebCore::FrameLoader::shouldTreatURLAsSrcdocDocument const):
(WebCore::FrameLoader::findFrameForNavigation):
(WebCore::FrameLoader::loadSameDocumentItem):
(WebCore::FrameLoader::loadDifferentDocumentItem):
(WebCore::FrameLoader::effectiveReferrerPolicy const):
(WebCore::FrameLoader::dispatchDidClearWindowObjectsInAllWorlds):
(WebCore::FrameLoader::dispatchDidClearWindowObjectInWorld):
(WebCore::FrameLoader::effectiveSandboxFlags const):
(WebCore::FrameLoader::didChangeTitle):
(WebCore::FrameLoader::dispatchDidCommitLoad):
(WebCore::FrameLoader::tellClientAboutPastMemoryCacheLoads):
(WebCore::FrameLoader::loadProgressingStatusChanged):
(WebCore::createWindow):
(WebCore::FrameLoader::switchBrowsingContextsGroup):
(WebCore::FrameLoader::shouldSuppressTextInputFromEditing const):
(WebCore::FrameLoader::protectedProvisionalDocumentLoader const):
* Source/WebCore/loader/FrameLoader.h:
* Source/WebCore/loader/ProgressTracker.h:
* Source/WebCore/page/Frame.cpp:
(WebCore::Frame::protectedWindowProxy const):
* Source/WebCore/page/Frame.h:
* Source/WebCore/page/LocalFrame.h:
* Source/WebCore/page/Page.cpp:
(WebCore::Page::checkedProgress):
(WebCore::Page::checkedProgress const):
* Source/WebCore/page/Page.h:
* Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundleFrame.cpp:
(WKBundleFrameClearOpener):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::m_historyItemClient):
* Source/WebKitLegacy/mac/WebView/WebFrame.mm:
(-[WebFrame _clearOpener]):

Canonical link: <a href="https://commits.webkit.org/269901@main">https://commits.webkit.org/269901@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d70adb3f4ee776f9d0c34eb6178a106a88dda2e3

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/24003 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/2114 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/25091 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/26144 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/22109 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/24275 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/3734 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/24486 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/22599 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/24247 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/1648 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/20747 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/26733 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/1404 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/21662 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/27895 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/21881 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/21950 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/25686 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/1339 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/19018 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/1358 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/1750 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/3060 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/1691 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rniwa commented on Source/WebCore/loader/FrameLoader.cpp:
```diff
@@ -250,29 +250,29 @@ struct ForbidPromptsScope : public PageLevelForbidScope {
     {
-        if (m_page)
-            m_page->forbidPrompts();
+        if (CheckedPtr page  = m_page.get())
```
Nit: there are two spaces before =.

rniwa commented on Source/WebCore/loader/FrameLoader.cpp:
```diff
@@ -250,29 +250,29 @@ struct ForbidPromptsScope : public PageLevelForbidScope {
     {
-        if (m_page)
-            m_page->allowPrompts();
+        if (CheckedPtr page  = m_page.get())
```
Ditto.

rniwa commented on Source/WebCore/loader/FrameLoader.cpp:
```diff
@@ -250,29 +250,29 @@ struct ForbidPromptsScope : public PageLevelForbidScope {
     {
-        if (m_page)
-            m_page->forbidSynchronousLoads();
+        if (CheckedPtr page  = m_page.get())
```
Ditto.

rniwa commented on Source/WebCore/loader/FrameLoader.cpp:
```diff
@@ -250,29 +250,29 @@ struct ForbidPromptsScope : public PageLevelForbidScope {
     {
-        if (m_page)
-            m_page->allowSynchronousLoads();
+        if (CheckedPtr page  = m_page.get())
```
Ditto.

rniwa commented on Source/WebCore/loader/FrameLoader.cpp:
```diff
@@ -525,56 +542,59 @@ void FrameLoader::submitForm(Ref<FormSubmission>&& submission)
+    // The frame may have started destruction, in which case we can't protect it.
+    RefPtr protectedFrame = m_frame->view() ? m_frame.ptr() : nullptr;
+
+    if (m_frame->document() && m_frame->document()->parser())
```
How about:
```cpp
if (RefPtr parser = m_frame->document() ? m_frame->document()->parser() : nullptr)
   parser->stopParsing();
```

rniwa commented on Source/WebCore/loader/FrameLoader.cpp:
```diff
@@ -525,56 +542,59 @@ void FrameLoader::submitForm(Ref<FormSubmission>&& submission)
     m_didCallImplicitClose = true; // don't want that one either
 
-    if (m_frame.document() && m_frame.document()->parsing()) {
+    if (m_frame->document() && m_frame->document()->parsing()) {
```
Can we store document in a local variable here?
```cpp
if (RefPtr document = m_frame->document(); document && document->parsing()) {
    finishedParsing();
    document->setParsing(false);
}
```

rniwa commented on Source/WebCore/loader/FrameLoader.cpp:
```diff
@@ -811,30 +838,31 @@ void FrameLoader::didBeginDocument(bool dispatch)
-    RefPtr protectedFrame { m_frame.view() ? &m_frame : nullptr };
+    RefPtr protectedFrame { m_frame->view() ? m_frame.ptr() : nullptr };
+
+    m_frame->injectUserScripts(UserScriptInjectionTime::DocumentEnd);
```
Can we used protectedFrame here?

cdumez replied:
protectedFrame can be null, see 2 lines above. This function may get called during the frame destruction.

rniwa commented on Source/WebCore/loader/FrameLoader.cpp:
```diff
@@ -864,7 +892,7 @@ bool FrameLoader::allChildrenAreComplete() const
 bool FrameLoader::allAncestorsAreComplete() const
 {
-    for (Frame* ancestor = &m_frame; ancestor; ancestor = ancestor->tree().parent()) {
+    for (Frame* ancestor = m_frame.ptr(); ancestor; ancestor = ancestor->tree().parent()) {
```
Can we use RefPtr here?

rniwa commented on Source/WebCore/loader/FrameLoader.cpp:
```diff
@@ -855,7 +883,7 @@ void FrameLoader::subresourceLoadDone(LoadCompletionType type)
 bool FrameLoader::allChildrenAreComplete() const
 {
-    for (auto* child = m_frame.tree().firstChild(); child; child = child->tree().nextSibling()) {
+    for (auto* child = m_frame->tree().firstChild(); child; child = child->tree().nextSibling()) {
```
Can we use RefPtr here?

rniwa commented on Source/WebCore/loader/FrameLoader.cpp:
```diff
@@ -1116,28 +1147,26 @@ void FrameLoader::resetMultipleFormSubmissionProtection()
-        if (!localFrame)
-            continue;
-        localFrame->document()->setFirstPartyForCookies(url);
+    for (Frame* frame = m_frame.ptr(); frame; frame = frame->tree().traverseNext(m_frame.ptr())) {
```
Can we use RefPtr here?

rniwa commented on Source/WebCore/loader/FrameLoader.cpp:
```diff
@@ -1116,28 +1147,26 @@ void FrameLoader::resetMultipleFormSubmissionProtection()
     RegistrableDomain registrableDomain(url);
-    for (Frame* frame = &m_frame; frame; frame = frame->tree().traverseNext(&m_frame)) {
-        auto* localFrame = dynamicDowncast<LocalFrame>(frame);
+    for (Frame* frame = m_frame.ptr(); frame; frame = frame->tree().traverseNext(m_frame.ptr())) {
```
Ditto.

rniwa commented on Source/WebCore/loader/FrameLoader.cpp:
```diff
@@ -1235,31 +1264,27 @@ bool FrameLoader::isComplete() const
-        if (!localDescendant)
-            continue;
-        localDescendant->navigationScheduler().startTimer();
+    for (auto* descendant = frame->tree().traverseNext(m_frame.ptr()); descendant; descendant = descendant->tree().traverseNext(m_frame.ptr())) {
```
Can we use RefPtr here?

rniwa commented on Source/WebCore/loader/FrameLoader.cpp:
```diff
@@ -1235,31 +1264,27 @@ bool FrameLoader::isComplete() const
-        if (!localFrame)
-            continue;
-        localFrame->loader().m_isComplete = false;
+    for (Frame* frame = m_frame.ptr(); frame; frame = frame->tree().parent()) {
```
Ditto.

rniwa commented on Source/WebCore/loader/FrameLoader.cpp:
```diff
@@ -2160,72 +2182,71 @@ void FrameLoader::commitProvisionalLoad()
-                continue;
-            targetFrames.append(*localChild);
+        targetFrames.append(frame);
+        for (auto* child = frame->tree().firstChild(); child; child = child->tree().traverseNext(frame.ptr())) {
```
Can we use RefPtr here?

rniwa commented on Source/WebCore/loader/FrameLoader.cpp:
```diff
@@ -2376,11 +2397,9 @@ void FrameLoader::closeOldDataSources()
-        if (!localChild)
-            continue;
-        localChild->loader().closeOldDataSources();
+    for (auto* child = m_frame->tree().firstChild(); child; child = child->tree().nextSibling()) {
```
Can we use RefPtr here?

rniwa commented on Source/WebCore/loader/FrameLoader.cpp:
```diff
@@ -2469,23 +2488,23 @@ void FrameLoader::setReplacing()
     // It's most likely that the last added frame is the last to load so we walk backwards.
-    for (auto* child = m_frame.tree().lastChild(); child; child = child->tree().previousSibling()) {
-        auto* localChild = dynamicDowncast<LocalFrame>(child);
+    for (auto* child = m_frame->tree().lastChild(); child; child = child->tree().previousSibling()) {
```
Can we use RefPtr here?

rniwa commented on Source/WebCore/loader/FrameLoader.cpp:
```diff
@@ -2614,57 +2633,57 @@ void FrameLoader::checkLoadCompleteForThisFrame()
             // If we're in the middle of loading multipart data, we need to restore the document loader.
             if (isReplacing() && !m_documentLoader.get())
-                    setDocumentLoader(m_provisionalDocumentLoader.get());
+                setDocumentLoader(m_provisionalDocumentLoader.copyRef());
```
Can we use `provisionalDocumentLoader` instead?

rniwa commented on Source/WebCore/loader/FrameLoader.cpp:
```diff
@@ -2848,30 +2864,28 @@ void FrameLoader::detachChildren()
-            continue;
-        childrenToDetach.append(*localChild);
+    childrenToDetach.reserveInitialCapacity(m_frame->tree().childCount());
+    for (auto* child = m_frame->tree().lastChild(); child; child = child->tree().previousSibling()) {
```
Can we use RefPtr here?

rniwa commented on Source/WebCore/loader/FrameLoader.cpp:
```diff
@@ -2882,64 +2896,60 @@ void FrameLoader::closeAndRemoveChild(LocalFrame& child)
-        if (!localFrame)
-            continue;
-        frames.append(*localFrame);
+    for (auto* frame = &m_frame->mainFrame(); frame; frame = frame->tree().traverseNext()) {
```
Ditto.

rniwa commented on Source/WebCore/loader/FrameLoader.cpp:
```diff
@@ -2882,64 +2896,60 @@ void FrameLoader::closeAndRemoveChild(LocalFrame& child)
-        if (!localFrame)
-            continue;
-        count += localFrame->document()->cachedResourceLoader().requestCount();
+    for (Frame* frame = m_frame.ptr(); frame; frame = frame->tree().traverseNext(m_frame.ptr())) {
```
Ditto.

rniwa commented on Source/WebCore/loader/FrameLoader.cpp:
```diff
@@ -3458,30 +3470,28 @@ void FrameLoader::scrollToFragmentWithParentBoundary(const URL& url, bool isNewN
-            continue;
-        targetFrames.append(*localChild);
+    targetFrames.append(m_frame.get());
+    for (auto* child = m_frame->tree().firstChild(); child; child = child->tree().traverseNext(m_frame.ptr())) {
```
Can we use RefPtr?

rniwa commented on Source/WebCore/loader/FrameLoader.cpp:
```diff
@@ -4031,17 +4045,17 @@ void FrameLoader::loadDifferentDocumentItem(HistoryItem& item, HistoryItem* from
 
-    if (CachedPage* cachedPage = BackForwardCache::singleton().get(item, m_frame.page())) {
-        auto documentLoader = cachedPage->documentLoader();
+    if (CachedPage* cachedPage = BackForwardCache::singleton().get(item, frame->page())) {
```
Can we use CheckedPtr here?

rniwa commented on Source/WebCore/page/LocalFrame.h:
```diff
@@ -155,6 +155,7 @@ class LocalFrame final : public Frame, public CanMakeCheckedPtr {
     FrameLoader& loader() { return m_loader.get(); }
     CheckedRef<const FrameLoader> checkedLoader() const;
     CheckedRef<FrameLoader> checkedLoader();
+
```
What's the deal with this blank line?

cdumez replied:
I was trying to separate the 4 loader getters from the rest. They already had a blank line before them. It thought it looked nicer this way.

cdumez committed:
Adopt smart pointers in FrameLoader
https://bugs.webkit.org/show_bug.cgi?id=263796

Reviewed by Ryosuke Niwa.

* Source/WebCore/html/HTMLFormElement.cpp:
(WebCore::HTMLFormElement::submitIfPossible):
* Source/WebCore/html/HTMLFrameOwnerElement.h:
(WebCore::Frame::protectedOwnerElement const):
* Source/WebCore/inspector/InspectorFrontendClientLocal.cpp:
(WebCore::InspectorFrontendClientLocal::openURLExternally):
* Source/WebCore/loader/FormState.cpp:
(WebCore::FormState::protectedSourceDocument const):
* Source/WebCore/loader/FormState.h:
* Source/WebCore/loader/FrameLoadRequest.cpp:
(WebCore::FrameLoadRequest::FrameLoadRequest):
* Source/WebCore/loader/FrameLoadRequest.h:
(WebCore::FrameLoadRequest::FrameLoadRequest):
* Source/WebCore/loader/FrameLoader.cpp:
(WebCore::ForbidPromptsScope::ForbidPromptsScope):
(WebCore::ForbidPromptsScope::~ForbidPromptsScope):
(WebCore::ForbidSynchronousLoadsScope::ForbidSynchronousLoadsScope):
(WebCore::ForbidSynchronousLoadsScope::~ForbidSynchronousLoadsScope):
(WebCore::FrameLoader::FrameProgressTracker::FrameProgressTracker):
(WebCore::FrameLoader::FrameProgressTracker::~FrameProgressTracker):
(WebCore::FrameLoader::FrameProgressTracker::progressStarted):
(WebCore::FrameLoader::FrameProgressTracker::progressCompleted):
(WebCore::FrameLoader::frame const):
(WebCore::FrameLoader::protectedFrame const):
(WebCore::FrameLoader::detachFromAllOpenedFrames):
(WebCore::FrameLoader::init):
(WebCore::FrameLoader::initForSynthesizedDocument):
(WebCore::FrameLoader::pageID const):
(WebCore::FrameLoader::frameID const):
(WebCore::FrameLoader::setDefersLoading):
(WebCore::FrameLoader::shouldUpgradeRequestforHTTPSOnly const):
(WebCore::FrameLoader::changeLocation):
(WebCore::FrameLoader::submitForm):
(WebCore::FrameLoader::stopLoading):
(WebCore::FrameLoader::stop):
(WebCore::FrameLoader::closeURL):
(WebCore::FrameLoader::didOpenURL):
(WebCore::FrameLoader::didExplicitOpen):
(WebCore::FrameLoader::clear):
(WebCore::FrameLoader::receivedFirstData):
(WebCore::FrameLoader::didBeginDocument):
(WebCore::FrameLoader::finishedParsing):
(WebCore::FrameLoader::allChildrenAreComplete const):
(WebCore::FrameLoader::allAncestorsAreComplete const):
(WebCore::FrameLoader::checkCompleted):
(WebCore::FrameLoader::checkCompletenessNow):
(WebCore::FrameLoader::checkCallImplicitClose):
(WebCore::FrameLoader::loadURLIntoChildFrame):
(WebCore::FrameLoader::loadArchive):
(WebCore::FrameLoader::outgoingReferrer const):
(WebCore::FrameLoader::outgoingOrigin const):
(WebCore::FrameLoader::checkIfFormActionAllowedByCSP const):
(WebCore::FrameLoader::setOpener):
(WebCore::FrameLoader::provisionalLoadStarted):
(WebCore::FrameLoader::updateFirstPartyForCookies):
(WebCore::FrameLoader::setFirstPartyForCookies):
(WebCore::FrameLoader::loadInSameDocument):
(WebCore::FrameLoader::completed):
(WebCore::FrameLoader::started):
(WebCore::FrameLoader::prepareForLoadStart):
(WebCore::FrameLoader::loadFrameRequest):
(WebCore::FrameLoader::isNavigationAllowed const):
(WebCore::FrameLoader::loadURL):
(WebCore::FrameLoader::defaultSubstituteDataForURL):
(WebCore::FrameLoader::load):
(WebCore::FrameLoader::loadWithDocumentLoader):
(WebCore::FrameLoader::reloadWithOverrideEncoding):
(WebCore::FrameLoader::reload):
(WebCore::FrameLoader::stopAllLoaders):
(WebCore::FrameLoader::stopForBackForwardCache):
(WebCore::FrameLoader::stopForUserCancel):
(WebCore::FrameLoader::setDocumentLoader):
(WebCore::FrameLoader::setPolicyDocumentLoader):
(WebCore::FrameLoader::setProvisionalDocumentLoader):
(WebCore::FrameLoader::setState):
(WebCore::FrameLoader::provisionalLoadFailedInAnotherProcess):
(WebCore::FrameLoader::commitProvisionalLoad):
(WebCore::FrameLoader::transitionToCommitted):
(WebCore::FrameLoader::closeOldDataSources):
(WebCore::FrameLoader::willRestoreFromCachedPage):
(WebCore::FrameLoader::open):
(WebCore::FrameLoader::subframeIsLoading const):
(WebCore::FrameLoader::subresourceCachePolicy const):
(WebCore::FrameLoader::handleLoadFailureRecovery):
(WebCore::FrameLoader::checkLoadCompleteForThisFrame):
(WebCore::FrameLoader::setOriginalURLForDownloadRequest):
(WebCore::FrameLoader::didReachLayoutMilestone):
(WebCore::FrameLoader::didFirstLayout):
(WebCore::FrameLoader::restoreScrollPositionAndViewStateSoon):
(WebCore::FrameLoader::didReachVisuallyNonEmptyState):
(WebCore::FrameLoader::detachChildren):
(WebCore::FrameLoader::closeAndRemoveChild):
(WebCore::FrameLoader::checkLoadComplete):
(WebCore::FrameLoader::numPendingOrLoadingRequests const):
(WebCore::FrameLoader::userAgent const):
(WebCore::FrameLoader::navigatorPlatform const):
(WebCore::FrameLoader::frameDetached):
(WebCore::FrameLoader::detachFromParent):
(WebCore::FrameLoader::updateRequestAndAddExtraFields):
(WebCore::FrameLoader::scheduleRefreshIfNeeded):
(WebCore::FrameLoader::loadPostRequest):
(WebCore::FrameLoader::loadResourceSynchronously):
(WebCore::FrameLoader::receivedMainResourceError):
(WebCore::FrameLoader::continueFragmentScrollAfterNavigationPolicy):
(WebCore::FrameLoader::shouldPerformFragmentNavigation):
(WebCore::FrameLoader::scrollToFragmentWithParentBoundary):
(WebCore::FrameLoader::shouldClose):
(WebCore::FrameLoader::dispatchUnloadEvents):
(WebCore::FrameLoader::dispatchBeforeUnloadEvent):
(WebCore::FrameLoader::executeJavaScriptURL):
(WebCore::FrameLoader::continueLoadAfterNavigationPolicy):
(WebCore::FrameLoader::continueLoadAfterNewWindowPolicy):
(WebCore::FrameLoader::loadedResourceFromMemoryCache):
(WebCore::FrameLoader::shouldInterruptLoadForXFrameOptions):
(WebCore::FrameLoader::shouldTreatURLAsSameAsCurrent const):
(WebCore::FrameLoader::shouldTreatURLAsSrcdocDocument const):
(WebCore::FrameLoader::findFrameForNavigation):
(WebCore::FrameLoader::loadSameDocumentItem):
(WebCore::FrameLoader::loadDifferentDocumentItem):
(WebCore::FrameLoader::effectiveReferrerPolicy const):
(WebCore::FrameLoader::dispatchDidClearWindowObjectsInAllWorlds):
(WebCore::FrameLoader::dispatchDidClearWindowObjectInWorld):
(WebCore::FrameLoader::effectiveSandboxFlags const):
(WebCore::FrameLoader::didChangeTitle):
(WebCore::FrameLoader::dispatchDidCommitLoad):
(WebCore::FrameLoader::tellClientAboutPastMemoryCacheLoads):
(WebCore::FrameLoader::loadProgressingStatusChanged):
(WebCore::createWindow):
(WebCore::FrameLoader::switchBrowsingContextsGroup):
(WebCore::FrameLoader::shouldSuppressTextInputFromEditing const):
(WebCore::FrameLoader::protectedProvisionalDocumentLoader const):
* Source/WebCore/loader/FrameLoader.h:
* Source/WebCore/loader/ProgressTracker.h:
* Source/WebCore/page/Frame.cpp:
(WebCore::Frame::protectedWindowProxy const):
* Source/WebCore/page/Frame.h:
* Source/WebCore/page/LocalFrame.h:
* Source/WebCore/page/Page.cpp:
(WebCore::Page::checkedProgress):
(WebCore::Page::checkedProgress const):
* Source/WebCore/page/Page.h:
* Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundleFrame.cpp:
(WKBundleFrameClearOpener):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::m_historyItemClient):
* Source/WebKitLegacy/mac/WebView/WebFrame.mm:
(-[WebFrame _clearOpener]):

Canonical link: https://commits.webkit.org/269901@main
